### ISSUE TITLE
[question]: How to use NotificationWillDisplayEvent

### ISSUE THREAD

**Issue by serolgames** [OneSignal/react-native-onesignal#1588]
### How can we help?

Hello,

in older version we listened to incoming notification with 
OneSignal.setNotificationWillShowInForegroundHandler

which is still in use in the official documentation...

But I found that we should use event listener NotificationWillDisplayEvent.

How to use it and read notification data ? 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
Hello @serolgames thanks for reaching out! The updated method is adding an event listener for [`foregroundWillDisplay`](https://documentation.onesignal.com/docs/mobile-sdk#handling-notifications-before-theyre-presented-when-the-app-is-in-the-foreground).

Let us know if you have any additional questions or concerns.

Thanks!

**Comment by serolgames** [comment]
Hello,

Thanks you !
It's working but I found another issue :(

I'm using the following code : 

    const onReceived = (event: any) => 
    {                    
        event.preventDefault()
        const additionalData = event.getNotification().additionalData

        if(/*my logic to display only if I want*/)
        {
            event.getNotification().display();
        }
    }

    useEffect(() =>
    {
        OneSignal.Notifications.addEventListener('foregroundWillDisplay', onReceived);

        return () => {
            console.log("removing event...")
            OneSignal.Notifications.removeEventListener('foregroundWillDisplay', onReceived);
        };
    }, []);

So basically, it's working. BUT, the event seems to never be removed, I get the log "removing event..." so removeEventListener is properly called but it's never getting removed. Is it a known issue ? Or a problem in my code ?

**Comment by mathbalduino** [comment]
It's still happening
